### DISCUSSION TITLE
Rhai Performance

### DISCUSSION THREAD

**Discussion by tryoxiss** [rhaiscript/rhai#827] [General]
I know Rhai is not meant to be blazingly fast--and it dosen't need to be. But it does mention being fast as a feature, and looking at the benchmarks it runs about 5x slower than python, which isn't great given python is generally considered quite a slow language.

**Comment by schungx** [comment]
Well, it depends on your angle, of course.  It is fast _enough_ for many application scenarios.  It is also faster than many scripting engines of similar architecture (i.e. AST-walker). Python is not an AST-walker (it compiles to bytecodes), so it is not by far the _slowest_ language...

If you're comparing it with, say, V8 or LuaJIT then there is no contest.  So being fast is a fuzzy concept...  you need to know how fast is fast enough.

And balance it off with simply embedding the fastest engine (but perhaps paying a heavy integration cost).  If you can embed V8, you should -- then you will be _really_ fast, and your users do not need to learn yet another obscure language.  But not every application makes sense to embed V8.

**Comment by schungx** [comment]
Closing this for now. Feel free to reopen if necessary.
